File: scripts/1_create_schema.py
import weaviate

w1 = weaviate.Client("http://localhost:8080")

# Create Author class
class_obj = {
    "class": "Author",
    "description": "Fantasy Writers",
    "properties": [
        {
            "name": "name",
            "dataType": ["string"],
            "description": "The author's name",
        },
        {
            "name": "website",
            "dataType": ["string"],
            "description": "Their website address",
        },
        {
            "name": "genres",
            "dataType": ["text[]"],
            "description": "Genres they write",
        },        
    ]
}
w1.schema.create_class(class_obj)

# Create Book class
class_obj = {
    "class": "Book",
    "description": "Popular books",
    "properties": [
        {
            "name": "title",
            "dataType": ["string"],
            "description": "Title of the book",
        },
        {
            "name": "written_by",
            "dataType": ["Author"],
            "description": "Author of the book",
        },        
    ]
}
w1.schema.create_class(class_obj)

File: scripts/1_create_schema.go
package main

import (
	"context"
	"fmt"

	"github.com/semi-technologies/weaviate-go-client/v4/weaviate"
	"github.com/semi-technologies/weaviate/entities/models"
	"github.com/semi-technologies/weaviate/entities/schema"
)

func main() {
	hostW1 := "localhost:8080"

	author := &models.Class{
		Class:       "Author",
		Description: "Fantasy Writers",
		Properties: []*models.Property{
			{
				Name:        "name",
				Description: "The author's name",
				DataType:    []string{string(schema.DataTypeString)},
			},
			{
				Name:        "website",
				Description: "Their website address",
				DataType:    []string{string(schema.DataTypeString)},
			},
			{
				Name:        "genres",
				Description: "Genres they write",
				DataType:    []string{string(schema.DataTypeTextArray)},
			},
		},
	}

	book := &models.Class{
		Class:       "Book",
		Description: "Popular books",
		Properties: []*models.Property{
			{
				Name:        "title",
				Description: "Title of the book",
				DataType:    []string{string(schema.DataTypeString)},
			},
			{
				Name:        "written_by",
				Description: "Author of the book",
				DataType:    []string{"Author"},
			},
		},
	}

	ctx := context.Background()
	client := weaviate.New(weaviate.Config{Scheme: "http", Host: hostW1})
	classCreator := client.Schema().ClassCreator()

	fmt.Println("\nCreating Schema - Author")
	errAuthor := classCreator.WithClass(author).Do(ctx)
	if errAuthor != nil {
		fmt.Println(errAuthor)
		return
	}
	fmt.Println("Created")

	fmt.Println("\nCreating Schema - Book")
	errBook := classCreator.WithClass(book).Do(ctx)
	if errBook != nil {
		fmt.Println(errBook)
		return
	}
	fmt.Println("Created")
}
